### PR TITLE
cmd/syncthing: Refactor out staticsServer (prev. embeddedStatic) a bit

### PR DESCRIPTION
**Pull request by calmh**
### Purpose

The purpose of this operation is to separate the serving of GUI assets a bit from the serving of the REST API. It's by no means complete. The end goal is something like a combined server type that embeds a statics server and an API server and wraps it in authentication and HTTPS and stuff, plus possibly a named pipe server that only provides the API and does not wrap in the same authentication etc.

I also remove some of the caching logic that we no longer fully trust.

### Testing

Tested manually by starting the GUI, switching themes, changing the index.html and observing that it takes effect through the STGUIASSETS setting.

### CODE REVIEW

File: cmd/syncthing/gui_statics.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		file = "index.html"
+	}
+
+	s.mut.RLock()
+	theme := s.theme
+	s.mut.RUnlock()
+
+	// Check for an override for the current theme.
+	if s.assetDir != "" {
+		p := filepath.Join(s.assetDir, s.theme, filepath.FromSlash(file))
```
Comment (AudriusButkevicius) [inline cmd/syncthing/gui_statics.go]: Should not be s.theme?

Comment (calmh) [inline cmd/syncthing/gui_statics.go]: Correct, will fix (was apparently broken before as well)

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `pending`. I'll wait until it goes green and then merge!

**Comment by st-review** [comment]
:ok_hand: Merged as 03a8027efca84409550d1d773fc433a6bef72638. Thanks, @calmh!

**Pull request closed without merging**